cyfronet-fid/marketplace
# [#1493] Create personalized search results

This PR closes #1493 issue that include creating two versions of recommended services panels.

### TODO:
- [x] Create a dedicated "Recommended results" section on the search result page.
- [x] The current implementation (for all users) should be based on the most popular resources from the most popular category.
- [x] Some users should see view according to Mockup1 and others according to Mockup2
- [x] Use our split configuration to configure the experiment for each implementation
- [x] Fine-tune CSS (@jarekzet )
  - [x] Add split gem helper for testing
  - [x] SimpleRecommender
  - [x] RecommenderSystemHelper
  - [x] ServiceRecommendations
  - [x] Adapt old services feature specs for new layout with recommendations 
- [x] Remove "Based on your search" string from the recommendation panel
- [x] Add possibility of disabling recommendations 
- [x] Fully handle A/B testing by using proper `ab_finished`  

### How to manually test (for reviewers):
Go to the `admin` -> `AB experiments` and try variants `v1`, `v2` of the `recommendation_panel` experiment and compare appearance of `/services` page with mockups (links to mockups available in the #1493 issue). Try to refresh page few times and try few sessions in incognito.
- If you are logged user you should always see the same variant even if you click recommended service or restart the session.
- If you aren't logged user and you don't store cookies (as in incognito mode, it simulates different users) you should see each of two variants of recommendation 50% probability.

Moreover there is `disabled` variant of experiments that is a control group without recommendations.

### Further work proposal:
- Turn on `RecommenderSystemHelper` and `ServiceRecommendations` specs (temporarily disabled because of unknown bug of `use_ab_test` method in specs)
- Refactor of translations in spec folder (?)
